include/core/simulation_control.h

```diff
@@ -348,6 +348,17 @@ class SimulationControl
     return time_step;
   }
 
+  /**
+   * @brief Get current iteration number.
+   *
+   * @return Pointer to the current iteration number.
+   */
+  unsigned int *
+  get_iteration_number()
+  {
+    return &iteration_number;
+  }
```

Comment:
This is not a good idea. You should not be keeping raw pointers like that around.
Instead keep a reference to the SimulationControl object and request the iteration number, but we do not use raw pointers in Lethe anymore...